plugins/vpc-bridge/config/k8s_connector.go

```diff
@@ -0,0 +1,209 @@
+
+	// Create named pipe, used to get output from k8s connector binary.
+	// pipeListener will accept connections in separate go routine. Output of binary is received via channel.
+	pipeListener, executionResultChan, err := createNamePipeForOutputFromK8sConnector()
+	if err != nil {
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: error creating named pipe %w", err)
+	}
+	defer pipeListener.Close()       // Defer closing named pipe listener.
+	defer close(executionResultChan) // Defer closing result channel.
+
+	namedPipePath := getPipeName(pipeListener)
+	log.Debugf("named piped: %s created", namedPipePath)
+
+	// Execute k8s connector binary with pod namespace, pod name and named pipe path.
+	err = executeK8sConnector(podNamespace, podName, namedPipePath)
+	if err != nil {
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: error executing k8s connector %w", err)
+	}
+
+	// Read output of k8s binary from named pipe.
+	var result k8sExecutionResult
+	select {
+	case result = <-executionResultChan:
+	case <-time.After(k8sExecutionTimeout):
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: timeout getting output of k8s connector")
+	}
+
+	if result.stderr != nil {
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: error in k8s connector %w", result.stderr)
+	}
+	podIP := result.stdout
+	log.Debugf("got pod IP: %s for pod %s in namespace %s", podIP, podName, podNamespace)
+
+	return podIP, nil
+}
+
+// executeK8sConnector executes aws-vpc-cni-k8s-connector binary and get pod IP over named pipe.
+// Execution logs from binary are stderr.
+func executeK8sConnector(podNamespace, podName, pipe string) error {
+	// Get Path to aws-vpc-cni-k8s-connector binary from env variable.
+	connectorBinaryPath := os.Getenv("AWS_VPC_CNI_K8S_CONNECTOR_BINARY_PATH")
+	// Use VPC CNI log level for k8s connector binary.
+	binaryLogLevel := os.Getenv("VPC_CNI_LOG_LEVEL")
+
+	// Prepare command to execute binary with required args.
+	cmd := exec.Command(connectorBinaryPath,
+		"-pod-name", podName, "-pod-namespace", podNamespace,
+		"-pipe", pipe, "-log-level", binaryLogLevel)
+	log.Debugf("Executing cmd: %s to get pod IP", cmd.String())
+
+	var errBytes, outputBytes bytes.Buffer
+	cmd.Stderr = &errBytes
+	cmd.Stdout = &outputBytes
+
+	// Run command
+	err := cmd.Run()
+	if err != nil || cmd.ProcessState.ExitCode() != 0 {
+		return fmt.Errorf("executeK8sConnector: error running connector binary %w with error: %s", err, errBytes.String())
+	}
+
+	log.Infof("logs from k8s connector binary...\n%s ...end of k8s connector binary logs", errBytes.String())
+
+	return nil
+}
+
+// createNamePipeForOutputFromK8sConnector creates named pipe.
+// Starts listening and reading named pipe in go routine.
+// Returns listener for named pipe and channel for receiving output from channel.
+func createNamePipeForOutputFromK8sConnector() (net.Listener, chan k8sExecutionResult, error) {
+	// Generate GUID random string.
+	g, err := guid.NewV4()
```

Comment:
You can use-
```
guid, err := windows.GenerateGUID()
fmt.Println(guid.String())
```
This is part of `golang.org/x/sys/windows`.

Then you don't need to import `github.com/Microsoft/go-winio/pkg/guid` just for this.

Reply:
done and tested